Top user profile navigation for small screens - migrate to React

mrkvon opened the pull request:
#### Proposed Changes

* Migrate top profile navigation for small screens to React.

![top-navigation-small](https://user-images.githubusercontent.com/7449720/68628647-7b3d7780-04e1-11ea-9f3b-aa572de01700.png)

#### Testing Instructions

* Go to somebody's profile
* Make the browser smaller
* Check that the top profile navigation hasn't changed in looks and behaviour.
  - especially check that the correct links are displayed and that they behave as they did before
    - own profile: Edit your profile
    - not a contact: Send a message | Add contact
    - some of my contacts: Send a message | Remove contact
    - a contact request I sent: Send a message | Delete contact request
    - a contact request I received: Send a message | Delete contact request
    - additionally, when not own profile && references are enabled: Write a reference


Work on #1019

simison commented on modules/contacts/client/components/RemoveContact.js:
```diff
@@ -32,10 +40,10 @@ export function RemoveContact({ t, contact, show, inProgress, onHide, onRemoveCo
       <div className="modal-content">
         <Modal.Header>
-          <button type="button" className="close" aria-hidden="true" onClick={onHide} ng-if="!removeContactModal.isLoading">&times;</button>
+          <button type="button" className="close" aria-hidden="true" onClick={onCancel} ng-if="!removeContactModal.isLoading">&times;</button>
```
Remember to remove `ng-if` Angular bits. :-)

mrkvon replied:
Thank you @simison! :+1:
Fixed in a98e3403d544775fdae6d8e06074382cdabc9a46 as part of #1121.

mrkvon commented on modules/users/client/controllers/profile.client.controller.js:
```diff
@@ -23,11 +23,22 @@ function ProfileController(
-  vm.hasConnectedAdditionalSocialAccounts = hasConnectedAdditionalSocialAccounts;
-  vm.isConnectedSocialAccount = isConnectedSocialAccount;
-  vm.socialAccountLink = socialAccountLink;
-  vm.isWarmshowersId = isWarmshowersId;
```
These are unrelated to the migration itself. Unused variables hanging around.

mrkvon commented:
I'll be grateful if we can get this old PR finished. I've rebased and fixed some bugs. Would be great if somebody can review and/or approve so we can merge. Thanks!

simison commented on modules/users/client/components/TopNavigationSmall.component.js:
```diff
@@ -0,0 +1,173 @@
+  onContactRemoved: PropTypes.func.isRequired,
+};
+
+function LabelWithTooltip({ id, label, tooltip }) {
```
I think it's fine to leave tooltips out from mobile navigation; they never get shown since there's no _hover_ on mobile devices.

<img width="236" alt="image" src="https://user-images.githubusercontent.com/87168/76560974-08ea9d00-64ab-11ea-81ff-9798ddbd2aec.png">

That'll simplify this a bit.

mrkvon replied:
Is the small screen unlikely to be used on desktop?
Also, the tooltip is present in the original code. i.e. this change will degrade/simplify the original.
@simison If you're happy with the above points, I'll proceed with the change. I would appreciate a confirmation.

simison replied:
I think we can consider everything under 768px as a touch screen, and very unlikely used on a desktop.

If it was on original code it's my bad. 😅 Let's clean out now that we have a chance while looking into it.

mrkvon replied:
Removed the tooltip in ad97da390e46b965c2a789f0d030eb87dd361388.

simison commented on modules/users/client/components/TopNavigationSmall.component.js:
```diff
@@ -0,0 +1,123 @@
+  userId,
+  onContactRemoved,
+}) {
+  // @TODO this hacky fix should be removed and the data should be consistent
```
This could warrant opening an issue as a follow up for some day.

mrkvon replied:
Maybe a reminder to review all those migration-dependent TODOs... seems there will be more as we progress.

I love Typescript, because it helps so much with fixing inconsistencies like this one...

mrkvon commented:
Thank you, @simison for the collaboration!
